src/jit/lsraarm64.cpp

```diff
@@ -957,6 +957,12 @@ void LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
 
     switch (compiler->getHWIntrinsicInfo(intrinsicID).form)
     {
+        case HWIntrinsicInfo::Sha1HashOp:
```

Comment:
In this method, presumably above where you are already separately handling the 2 and 3 operand cases, you need to set op2 (and op3 if present) as delay free, so that they won't be the same as the target register. Otherwise, you can't copy op1 to the target without possibly trashing another operand.
This may complicate the code a bit (though it will be potentially a bit simpler after #16517.
If none of the operands can be contained (i.e. used from memory, or an immediate that can be directly encoded), then instead of calling `GetOperandInfo` you would call `getLocationInfo` on those operands, e.g.
```
LocationInfoListNode* op2Info = getLocationInfo(op2);
op2Info->info.isDelayFree = true;
info->hasDelayFreeSrc = true;
```
And similar for op3 if present. If either of these can be contained, you need to handle those cases. You could possibly use `CheckAndSetDelayFree` from lsraxarch.cpp, moving it to lsrabuild.cpp. I hope that's not necessary as that code is actually going away and being handled another way in #16517.

Reply:
@CarolEidt @sdmaclea getLocationInfo() causing assertion failure here . Below seems to work as intended . Does this look OK?

       case HWIntrinsicInfo::Sha1HashOp:
            assert(info->srcCount == 3);
            info->setInternalCandidates(this, RBM_ALLFLOAT);
            info->internalFloatCount = 1;
            if (!op2->isContained())
            {
                LocationInfoListNode* op2Info = useList.Begin()->Next();
                op2Info->info.isDelayFree     = true;
                LocationInfoListNode* op3Info = op2Info->Next();
                op3Info->info.isDelayFree     = true;
                info->hasDelayFreeSrc         = true;
            }
            break;
        case HWIntrinsicInfo::SimdTernaryRMWOp:
            assert(info->srcCount == 3);
            if (!op2->isContained())
            {
                LocationInfoListNode* op2Info = useList.Begin()->Next();
                op2Info->info.isDelayFree     = true;
                LocationInfoListNode* op3Info = op2Info->Next();
                op3Info->info.isDelayFree     = true;
                info->hasDelayFreeSrc         = true;
            }
            break;

Reply:
> getLocationInfo() causing assertion failure here . Below seems to work as intended . Does this look OK?

Yes, my mistake. I had missed the fact that the operand info is retrieved above (n the calls to `GetOperandInfo()`). 
However, here you are assuming that there is a one-to-one match between the `LocationInfoListNode`s and the operands, which is true as long as the operands are not contained. You check `op2`, but you should add an `assert(!op3->isContained())` prior to accessing its info.